plugins/azure_sentinel/icon_azure_sentinel/util/api.py

```diff
@@ -0,0 +1,275 @@
+        self.logger.info("Getting token from: " + formatted_endpoint)
+
+        request = requests.request("POST", formatted_endpoint, data=data)
+        self.logger.info("Authentication request status: " + str(request.status_code))
+
+        if request.status_code != 200:
+            self.logger.error(request.text)
+            raise PluginException(
+                cause="Unable to authorize against Microsoft graph API.",
+                assistance="The application may not be authorized to connect "
+                "to the Microsoft Graph API. Please verify your connection information is correct and contact your "
+                "Azure administrator.",
+                data=request.text,
+            )
+        token = request.json().get("access_token")
+        self.time_ago = time.time()
+        self._auth_token = token
+        self.logger.info(f"Authentication Token: ****************{self.auth_token[-5:]}")
+        return token
+
+
+class AzureSentinelClient(AzureClient):
+    def __init__(self, logger: Any, tenant_id: str, client_id: str, client_secret: str):
+        super(AzureSentinelClient, self).__init__(logger, tenant_id, client_id, client_secret)  # noqa
+        self.headers = {"Content-Type": "application/json", "Authorization": f"Bearer {self.auth_token}"}
+
+    def _call_api(
+        self, method: str, url: str, headers: dict, payload: Optional[dict] = None, params: Optional[dict] = None
+    ) -> Tuple[int, Dict]:
+        """_call_api. Send a http call to a given endpoint.
+
+        :param method: Http method to be called.
+        :type method: str
+        :param url: Url to be called.
+        :type url: str
+        :param headers: Headers to be sent in the http call.
+        :type headers: dict
+        :param payload: Payload to be sent in the http call.
+        :type payload: Optional[dict]
+        :rtype: Union[Dict, int]
+        """
+
+        try:
+            data = json.dumps(payload) if payload else None
+            if params:
+                payload_str = urllib.parse.urlencode(params, safe="$")
+                self.logger.info(payload_str)
+            else:
+                payload_str = ""
+            response = requests.request(method, url, headers=headers, data=data, params=payload_str)
+            response.raise_for_status()
+            if response.headers.get("content-type") == "application/json; charset=utf-8":
+                return response.status_code, response.json()
+            return response.status_code, {}
+
+        except json.decoder.JSONDecodeError as error:
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=error)
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code == 400:
+                raise PluginException(
+                    cause="Comibination of input arguments is incorrect", assistance=error.response.json().get("error")
```

Comment:
"Combination of input arguments is incorrect "

Reply:
What happens is that some combination of json fields are incorrect, BUT Microsoft's doc doesn't say a priori which ones. There are so many possible combinations and so many sensible combinations that it's impossible to check them all.  `assistance=error.response.json().get("error")` this message should give the customer _some_ idea how to change their input to satisfy Microsoft's requirements.

Reply:
I believe it was just a typo.

```suggestion
                    cause="Combination of input arguments is incorrect", assistance=error.response.json().get("error")
```

Reply:
Yeah true, thanks!